src/lib/shadow-shim-helper-rs/src/syscall_types.rs

```diff
@@ -99,6 +99,16 @@ impl<T> ForeignPtr<T> {
         let val = self.val;
         Self::new_with_type_inference(val - count * std::mem::size_of::<T>())
     }
+
+    #[inline]
+    pub fn into_raw(self) -> *const T {
+        self.val as *const T
+    }
+
+    #[inline]
+    pub fn into_raw_mut(self) -> *mut T {
+        self.val as *mut T
+    }
```

Comment:
nit: maybe doc comment to remind the caller to make sure the `ForeignPtr` is valid in this address space, otherwise it creates a dangling pointer